File: codegenerator/benchmarks/java/TrafficLightExec.java
import de.hhu.stups.btypes.BInteger;

public class TrafficLightExec {

    private TrafficLight TrafficLight = new TrafficLight();



    private BInteger counter;

    private boolean initialized = false;

    public void initialize() {
        if(initialized) {
            throw new RuntimeException("Machine is already initialized");
        }
        TrafficLight.initialize();
        counter = (BInteger) new BInteger("0");
        initialized = true;
    }

    public void simulate() {
        if(!initialized) {
            throw new RuntimeException("Machine was not initialized");
        }
        while((counter.less(new BInteger("500000"))).booleanValue()) {
            this.TrafficLight.cars_ry();
            this.TrafficLight.cars_g();
            this.TrafficLight.cars_y();
            this.TrafficLight.cars_r();
            this.TrafficLight.peds_g();
            this.TrafficLight.peds_r();
            counter = (BInteger) counter.plus(new BInteger("1"));
        }
    }

    public static void main(String[] args) {
        long start = System.nanoTime();
        TrafficLightExec exec = new TrafficLightExec();
        long end = System.nanoTime();
        System.out.println("Loading: " + (end - start));
        start = System.nanoTime();
        exec.initialize();
        end = System.nanoTime();
        System.out.println("Initialisation: " + (end - start));
        start = System.nanoTime();
        exec.simulate();
        end = System.nanoTime();
        System.out.println("Execution: " + (end - start));
    }

}

File: codegenerator/benchmarks/cpp/TrafficLightExec.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BInteger.cpp"
#include "BBoolean.cpp"
#include "TrafficLight.cpp"

#ifndef TrafficLightExec_H
#define TrafficLightExec_H

using namespace std;

class TrafficLightExec {



    private:

        TrafficLight TrafficLight;


        BInteger counter;

        bool initialized = false;

    public:

        TrafficLightExec(){}

        void initialize() {
            if(initialized) {
                throw runtime_error("Machine is already initialized");
            }
            TrafficLight.initialize();
            counter = static_cast<BInteger >((BInteger(0)));
            initialized = true;
        }

        void simulate() {
            if(!initialized) {
                throw runtime_error("Machine was not initialized");
            }
            while((counter.less((BInteger(500000)))).booleanValue()) {
                this->TrafficLight.cars_ry();
                this->TrafficLight.cars_g();
                this->TrafficLight.cars_y();
                this->TrafficLight.cars_r();
                this->TrafficLight.peds_g();
                this->TrafficLight.peds_r();
                counter = static_cast<BInteger >(counter.plus((BInteger(1))));
            }
        }

};

int main() {
    clock_t start,finish;
    double time;
    TrafficLightExec exec;
    exec.initialize();
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

